src/couch_replicator_httpc.erl

```diff
@@ -162,7 +167,11 @@ process_stream_response(ReqId, Worker, HttpDb, Params, Callback) ->
     receive
     {ibrowse_async_headers, ReqId, Code, Headers} ->
         case list_to_integer(Code) of
+        429 ->
+            Timeout = couch_replicator_ratelimiter:limit(),
```

Comment:
Agree. couch_replicator_rate_limiter would be better.

The timeout vs limit is because the rate limiter doesn't know how the calling code will use the limit. It is rather generic. Calling code uses the limit as a timeout for change feeds, but it is also used below to know when to kill the replication job when the limit is reached.

Maybe `:max_period()`?